WordPress/gutenberg
# Block Editor: Update BEM syntax to CSS modifer guidelines

Previously: https://github.com/WordPress/gutenberg/pull/17846#discussion_r368140643, https://github.com/WordPress/gutenberg/pull/16790#discussion_r368149717

This pull request seeks to update incorrect usage of BEM modifier syntax to use the `is-`-based modifier syntax as recommended in the [CSS naming guidelines](https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#naming).

This impacts two components:

- `LinkControl`
- `ResponveBlockControl`

In the latter case, the implementation has been revised as mentioned at https://github.com/WordPress/gutenberg/pull/16790#discussion_r368149717 to unify the application of a single `is-responsive` modifier class.

**Open Question:** There was a `hidden` attribute included previously in `ResponsiveBlockControl` which would never be applied given how the logic flowed. The changes here seek only to preserve the current behavior. If it needs to be applied, I can restore it as appropriate.

**Testing Instructions:**

Repeat testing instructions from #17846 and #16790, verifying that there are no regressions.

Ensure tests pass:

```
npm test
```
